test/log.spec.js

```diff
@@ -889,133 +881,130 @@ apis.forEach((IPFS) => {
-        assert.equal(log2.values[1].payload, 'entryA8')
-        assert.equal(log2.values[2].payload, 'entryA9')
-        assert.equal(log2.values[3].payload, 'entryA10')
+        assert.strictEqual(log2.id, data.heads[0].id)
+        assert.strictEqual(log2.length, 4)
+        assert.strictEqual(log2.values[0].payload, 'entryC0')
+        assert.strictEqual(log2.values[1].payload, 'entryA8')
+        assert.strictEqual(log2.values[2].payload, 'entryA9')
+        assert.strictEqual(log2.values[3].payload, 'entryA10')
 
         let log3 = await Log.fromEntry(ipfs, data.heads, 7)
-        assert.equal(log3.id, data.heads[0].id)
-        assert.equal(log3.length, 7)
-        assert.equal(log3.values[0].payload, 'entryB5')
-        assert.equal(log3.values[1].payload, 'entryA6')
-        assert.equal(log3.values[2].payload, 'entryC0')
-        assert.equal(log3.values[3].payload, 'entryA7')
-        assert.equal(log3.values[4].payload, 'entryA8')
-        assert.equal(log3.values[5].payload, 'entryA9')
-        assert.equal(log3.values[6].payload, 'entryA10')
+        assert.strictEqual(log3.id, data.heads[0].id)
+        assert.strictEqual(log3.length, 7)
+        assert.strictEqual(log3.values[0].payload, 'entryB5')
+        assert.strictEqual(log3.values[1].payload, 'entryA6')
+        assert.strictEqual(log3.values[2].payload, 'entryC0')
+        assert.strictEqual(log3.values[3].payload, 'entryA7')
+        assert.strictEqual(log3.values[4].payload, 'entryA8')
+        assert.strictEqual(log3.values[5].payload, 'entryA9')
+        assert.strictEqual(log3.values[6].payload, 'entryA10')
       })
 
       it('onProgress callback is fired for each entry', async () => {
-        const log1 = new Log(ipfs, 'A')
         let items1 = []
         const amount = 100
-        for(let i = 1; i <= amount; i ++) {
+        for (let i = 1; i <= amount; i++) {
           const prev1 = last(items1)
           const n1 = await Entry.create(ipfs, null, 'A', 'entryA' + i, [prev1])
           items1.push(n1)
         }
 
         let i = 0
-        let prevDepth = 0
         const callback = (hash, entry, depth) => {
-          assert.notEqual(entry, null)
-          assert.equal(hash, items1[items1.length - i - 1].hash)
-          assert.equal(entry.hash, items1[items1.length - i - 1].hash)
-          assert.equal(entry.payload, items1[items1.length - i - 1].payload)
-          assert.equal(depth - 1, i)
-
-          i ++
-          prevDepth = depth
+          assert.notStrictEqual(entry, null)
+          assert.strictEqual(hash, items1[items1.length - i - 1].hash)
+          assert.strictEqual(entry.hash, items1[items1.length - i - 1].hash)
+          assert.strictEqual(entry.payload, items1[items1.length - i - 1].payload)
+          assert.strictEqual(depth - 1, i)
+
+          i++
         }
 
-        const a = await Log.fromEntry(ipfs, last(items1), -1, [], callback)
+        await Log.fromEntry(ipfs, last(items1), -1, [], callback)
```

Comment:
Noticed here that we're actually not testing that the callback gets called 😄 What can happen here is that the callback gets *never* called and the test passes as there's no assert after `Log.fromEntry`.

We don't need to fix this now but should keep in mind to fix it.

Reply:
Handled here https://github.com/orbitdb/ipfs-log/pull/172